src/services/cartService.js

```diff
@@ -0,0 +1,18 @@
+import { CART_API } from "../constants";
+
+function OutOfStockException(message) {
+  this.message = message;
+}
+export function addToCart(formData) {
+  return fetch(CART_API, {
+    method: "POST",
+    headers: {
+      Authorization: "Token " + localStorage.getItem("token"),
+    },
+    body: formData,
+  }).then((res) => {
+    console.log(res);
+    if (res.status === 200) return res.json();
+    throw new OutOfStockException("Product is out of stock");
+  });
+}
```

Comment:
You can't throw `OutOfStockException` exception when the response status code is not 200, there can be other possible errors too like 500 and so on. So, handle this case specifically and all other possible cases as well.

Reply:
Done